src/main/ruby/core/array.rb

```diff
@@ -1311,6 +1311,27 @@ def recursively_flatten(array, out, max_levels = -1)
 
   private :recursively_flatten
 
+  def sort(&block)
+    sort_impl(block)
+  end
+
+  # Currently primitives do not support a block argument
+  private def sort_impl(block)
```

Comment:
I realize this is private, but I'd rather move this out to an `array_operations.rb` file.

Reply:
There are many private methods in array, especially around sorting (`mergesort_block`, etc).
I think it's out of scope of this PR to clean that up.